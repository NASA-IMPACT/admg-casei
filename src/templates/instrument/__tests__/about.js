import React from "react"
import { create } from "react-test-renderer"

import About from "../about"

const instrumentTypes = [
  {
    id: "test",
    longname: "test",
  },
  {
    id: "test2",
    longname: "test2",
  },
]
const radiometricFrequency = "test"
const temporalResolution = "test"
const spatialResolution = "test"
const calibration = "test"
const measurementRegions = [
  {
    id: "test",
    longname: "test",
  },
  {
    id: "test2",
    longname: "test2",
  },
]
const gcmdPhenomenas = [
  {
    term: "test",
    topic: "test",
    variable_1: "test",
    variable_2: "test",
    variable_3: "test",
  },
]
const instrumentDoi = "test"
const instrumentManufacturer = "test"
const fundingSource = "test"
const leadInvestigator = "test"
const technicalContact = "test"
const overviewPublication = "test"
const repositories = [
  {
    id: "test",
    longname: "test",
  },
  {
    id: "test2",
    longname: "test2",
  },
]

describe("About Section", () => {
  it("displays content", () => {
    const tree = create(
      <About
        instrumentTypes={instrumentTypes}
        radiometricFrequency={radiometricFrequency}
        temporalResolution={temporalResolution}
        spatialResolution={spatialResolution}
        calibration={calibration}
        measurementRegions={measurementRegions}
        gcmdPhenomenas={gcmdPhenomenas}
        instrumentDoi={instrumentDoi}
        instrumentManufacturer={instrumentManufacturer}
        fundingSource={fundingSource}
        leadInvestigator={leadInvestigator}
        technicalContact={technicalContact}
        overviewPublication={overviewPublication}
        repositories={repositories}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
