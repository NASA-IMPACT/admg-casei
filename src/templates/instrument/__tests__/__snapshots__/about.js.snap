// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Section displays content 1`] = `
.c3 {
  display: grid;
  grid-gap: 0.5rem;
  grid-template-columns: min-content 1fr;
  grid-auto-rows: auto;
}

.c3 dt {
  width: 12rem;
  color: #9E9E9E;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  margin-top: 2rem;
  position: relative;
}

.c1 {
  position: absolute;
  top: -80px;
}

.c2 {
  grid-column: 1 / span 8;
}

.c4 {
  grid-column: 10 / span 3;
}

<section
  className="c0"
  data-cy="unique-id-section"
>
  <div
    className="c1"
    id="unique-id"
  />
  <div
    style={
      Object {
        "alignSelf": "end",
        "gridColumn": "1 / span 7",
      }
    }
  >
    <a
      href="#unique-id"
    >
      <h2>
        Instrument Details
      </h2>
    </a>
  </div>
  <div
    className="c2"
  >
    <section
      data-cy="instrument-definition-list"
      style={
        Object {
          "padding": "1rem",
        }
      }
    >
      <dl
        className="c3"
      >
        <dt>
          <label>
            Instrument Type
          </label>
        </dt>
        <dd>
          test, test2
        </dd>
        <dt>
          <label>
            Measurement/Variables
          </label>
        </dt>
        <dd>
          test, test, test, test, test
        </dd>
        <dt>
          <label>
            Measurement Regions
          </label>
        </dt>
        <dd>
          test, test2
        </dd>
        <dt>
          <label>
            Temporal Resolution
          </label>
        </dt>
        <dd>
          test
        </dd>
        <dt>
          <label>
            Spatial Resolution
          </label>
        </dt>
        <dd>
          test
        </dd>
        <dt>
          <label>
            Measurement Frequency
          </label>
        </dt>
        <dd>
          test
        </dd>
        <dt>
          <label>
            Calibration Details
          </label>
        </dt>
        <dd>
          <a
            data-cy="calibration-doi-link"
            href="http://dx.doi.org/10.5067/calibration"
            rel="noopener noreferrer"
            style={
              Object {
                "color": "#aac9ff",
              }
            }
            target="_blank"
          >
            http://dx.doi.org/10.5067/calibration
          </a>
        </dd>
        <dt>
          <label>
            Online Information
          </label>
        </dt>
        <dd>
          <ul
            style={
              Object {
                "margin": "0",
              }
            }
          >
            <li
              style={
                Object {
                  "listStyle": "none",
                }
              }
            >
              <a
                data-cy="online-information-link"
                href="http://www.example.com"
                rel="noopener noreferrer"
                style={
                  Object {
                    "color": "#aac9ff",
                  }
                }
                target="_blank"
              >
                http://www.example.com
              </a>
            </li>
          </ul>
        </dd>
      </dl>
    </section>
  </div>
  <div
    className="c4"
  >
    <div
      data-cy="instrument-background"
    >
      <div
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <label
          data-cy="lead-investigator-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
            }
          }
        >
          Lead Investigator
        </label>
        test
      </div>
      <div
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <label
          data-cy="technical-contact-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
            }
          }
        >
          Technical Contact
        </label>
        test
      </div>
      <div
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <label
          data-cy="instrument-manufacturer-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
            }
          }
        >
          Instrument Maker
        </label>
        test
      </div>
      <div
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <label
          data-cy="funding-source-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
            }
          }
        >
          Funding Source
        </label>
        test
      </div>
      <div
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <p
          style={
            Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            }
          }
        >
          DOI:
           
          <a
            data-cy="instrument-doi-link"
            href="http://dx.doi.org/10.5067/instrument"
            rel="noopener noreferrer"
            style={
              Object {
                "color": "#aac9ff",
              }
            }
            target="_blank"
          >
            http://dx.doi.org/10.5067/instrument
          </a>
        </p>
      </div>
      <div
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <label
          data-cy="data-locations-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
            }
          }
        >
          Data Locations
        </label>
        N/A
      </div>
      <div
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <label
          data-cy="repositories-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
            }
          }
        >
          Repositories
        </label>
        <ul
          style={
            Object {
              "margin": "0",
            }
          }
        >
          <li
            data-cy="repository"
            style={
              Object {
                "listStyle": "none",
                "paddingBottom": "4px",
              }
            }
          >
            <p>
              test
            </p>
          </li>
          <li
            data-cy="repository"
            style={
              Object {
                "listStyle": "none",
                "paddingBottom": "4px",
              }
            }
          >
            <p>
              test2
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
